package/MDAnalysis/topology/PDBParser.py

```diff
@@ -303,6 +308,30 @@ def _parseatoms(self):
         masses = guess_masses(atomtypes)
         attrs.append(Masses(masses, guessed=True))
 
+        # Guessed attributes
+        # Need to pull elements from Atom names
+        # Similar to the check for atomtypes function
+        if not any(elements):
+            warnings.warn("Element information missing. Guessing elements from atom names : {}".format(names))
+            elements = guess_types(names)
+            attrs.append(Elements(elements, guessed=True))
+        else:
+            elements = guess_types(elements)
+            uni_elements, indices = np.unique(elements, return_inverse=True)
+
+            if any(np.isin(uni_elements, '')):
+                indexlist = [ j for j in range(len(indices)) if indices[j] == 0 ]
+                # This would be 0 always as guess_types would return a set
+                # of alphabets and empty string, and empty string will be
+                # sorted to be a index 0.
```

Comment:
This is a bit convoluted. You're right that the empty string will always be sorted to be first, so you could just check `if uni_elements[0] == ''`. However what I had in mind was more checking the unique atom names. So you can get a masking array of ``missing = elements==''``, use that to choose the names you want to look at with ``names_to_guess_from = names[missing]``, and apply `np.unique` to *that* array. (note: you may need to convert `elements` and `names` into numpy arrays first.)